packages/ai/ai/src/Prompt.ts

```diff
@@ -1659,6 +1662,84 @@ export const fromResponseParts = (parts: ReadonlyArray<Response.AnyPart>): Promp
           }))
           break
         }
+        case "tool": {
```

Comment:
Do we really need to add these new parts to the `AnyPart` interface? To me, it feels like these new parts are more of a utility thing, where you have a streaming response coming in and want to convert between the actual wire format and this new "accumulated" format. 

I am thinking it is desirable to keep the two separated. What are your thoughts? Are there use cases I'm not considering here?